x-pack/plugins/fleet/common/experimental_features.ts

```diff
@@ -11,7 +11,7 @@ export type ExperimentalFeatures = typeof allowedExperimentalValues;
  * A list of allowed values that can be used in `xpack.fleet.enableExperimental`.
  * This object is then used to validate and parse the value entered.
  */
-export const allowedExperimentalValues = Object.freeze({
+export const allowedExperimentalValues = Object.freeze<Record<string, boolean | undefined>>({
```

Comment:
`Object.freeze` also "freezes" the type definition to an extent, so because we had `foo: true` for each property of this object, the `value` of each property was frozen as type `true` rather than `boolean`. So, when we went to reassign it in the `partial` version of this object below, the type was "stuck" on `true | undefined` (`partial` basically just adds `| undefined` to every property of an object). Providing a looser typed generic here allows us to mutate the object below. 🤓